### PR TITLE
Clarify initial route usage with cached engines.

### PR DESCRIPTION
**Pull request by matthew-carroll**
Clarify initial route usage with cached engines.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@sfshaza2 the question of using an initial route with a cached engine applies to both `FlutterActivity` and `FlutterFragment` in identical ways, so it's identical text. I could have used just the term `Activity` in the 1st one and just `Fragment` in the 2nd one, but since everything else is identical, I figured best to make them the same so any future updates can be copied and pasted with no additional work.

Also, it looks like we now have some kind of enforced review process. Can you explicitly add your approval via the GitHub button so that I'll be allowed to merge?

**Comment by matthew-carroll** [comment]
@RedBrogdon I added you as a reviewer to this in case you have a moment to press the GitHub approve button. I'm currently blocked from merging until someone does that...

**Comment by legalcodes** [comment]
@matthew-carroll feel free to reach out to me as well if the situation arises again, happy to help out.

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 1]
```diff
@@ -311,6 +311,85 @@ pre-warmed `FlutterEngine` based on the ID given to `withCachedEngine()`.
 By providing a pre-warmed `FlutterEngine`, as previously shown, your app renders the
 first Flutter frame as quickly as possible.
 
+#### Initial route with a cached engine
+
+The concept of an initial route is available when configuring a 
+`FlutterActivity` or a `FlutterFragment` with a new `FlutterEngine`. However,
+`FlutterActivity` and `FlutterFragment` do not offer the concept of an initial
```
Comment (legalcodes) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: Nit:
```suggestion
`FlutterActivity` and `FlutterFragment` don't offer the concept of an initial
```
One reason that such contractions are useful: it's sometimes easy for a reader to miss the word "not," whereas it's harder to misread "don't" as "do."

See https://developers.google.com/style/contractions

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 2]
```diff
@@ -311,6 +311,85 @@ pre-warmed `FlutterEngine` based on the ID given to `withCachedEngine()`.
...
+route when using a cached engine. This is because a cached engine is expected
+to already be running Dart code, which means it's too late to configure the
+initial route.
+
+Developers that would like their cached engine to begin with a custom
+initial route can configure their cached `FlutterEngine` to use a custom initial
+route just before executing the Dart entrypoint. The following example
+demonstrates the use of an initial route with a cached engine.
```
Comment (legalcodes) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: Nit: 
```suggestion
demonstrates the use of an initial route with a cached engine:
```

File: src/docs/development/add-to-app/android/add-flutter-fragment.md [hunk 3]
```diff
@@ -311,6 +311,85 @@ pre-warmed `FlutterEngine` based on the ID given to `withCachedEngine()`.
...
+{% endsamplecode %}
+
+By setting the initial route of the navigation channel, the associated
+`FlutterEngine` displays the desired route upon initial execution of the
+`runApp()` Dart function.
+
+Changing the initial route property of the navigation channel after the initial
+execution of `runApp()` has no effect. Developers who would like to use
+the same `FlutterEngine` between different `Activity`s and `Fragment`s and
+switch the route between those displays need to setup a method channel and
```
Comment (legalcodes) [inline src/docs/development/add-to-app/android/add-flutter-fragment.md]: Nit:
```suggestion
switch the route between those displays need to set up a method channel and
```

**Comment by matthew-carroll** [comment]
Thanks @legalcodes. I'm gonna add you to another website PR.

**Comment by legalcodes** [comment]
Sure, np

**Comment by legalcodes** [comment]
Also, if this has urgency no need to worry about the nits. We can track them and make the updates at a later time.

**Comment by sfshaza2** [comment]
@matthew-carroll, @legalcodes, I wouldn't have approved this PR. Rather than adding identical text to two pages, I would have refactored it by creating a _foo.md file and including it into both places. Otherwise it's difficult to maintain. You can see this sort of thing done in the /src/docs/get-started/install directory.

**Comment by legalcodes** [comment]
@sfshaza2 Addressed in https://github.com/flutter/website/pull/3576

**Pull request merged**